lib/apiservers/engine/backends/container_proxy.go

```diff
@@ -135,9 +137,13 @@ const (
 	ContainerExited  = "exited"
 )
 
+// used by other engine components
+var containerProxy *ContainerProxy
+
 // NewContainerProxy creates a new ContainerProxy
 func NewContainerProxy(plClient *client.PortLayer, portlayerAddr string, portlayerName string) *ContainerProxy {
-	return &ContainerProxy{client: plClient, portlayerAddr: portlayerAddr, portlayerName: portlayerName}
+	containerProxy = &ContainerProxy{client: plClient, portlayerAddr: portlayerAddr, portlayerName: portlayerName}
+	return containerProxy
```

Comment:
if we are going to use it like this, then I prefer the singleton method we use in tether

https://github.com/vmware/vic/blob/957f703a2ff47d5975a5282709bbdca62f363bbd/cmd/tether/attach.go#L39
https://github.com/vmware/vic/blob/957f703a2ff47d5975a5282709bbdca62f363bbd/cmd/tether/attach.go#L80

otherwise LGTM